lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -456,6 +460,14 @@ class PartitionOpTranslator {
     auto getSourceArg = [&](unsigned i) {
       if (i < sourceApply->getArgs()->size())
         return sourceApply->getArgs()->getExpr(i);
+      llvm::dbgs() << "While processing SIL instruction ";
+      applyInst->dump();
+      llvm::dbgs() << "attempted to look up arg number "
+                   << i
+                   << " when only "
+                   << sourceApply->getArgs()->size()
+                   << " were present in the AST node: ";
+      sourceApply->dump();
       assert(false && "SIL instruction has too many arguments for"
                       " corresponding AST node");
       return (Expr *)nullptr;
```

Comment:
Hm, I suggest doing an `llvm_unreachable` instead of returning nullptr here. Since dump is not available in non-debug builds, you'll also want to guard the message with an `#if` like this:

```suggestion
      if (i < sourceApply->getArgs()->size())
        return sourceApply->getArgs()->getExpr(i);
        
#ifndef NDEBUG        
      llvm::dbgs() << "While processing SIL instruction ";
      applyInst->dump();
      llvm::dbgs() << "attempted to look up arg number "
                   << i
                   << " when only "
                   << sourceApply->getArgs()->size()
                   << " were present in the AST node: ";
      sourceApply->dump();
#endif
      llvm_unreachable("SendNonSendable: SIL instruction has too many arguments for corresponding AST node");
```